<launch>
  <arg name="debug" default="true"/>
  <arg name="detector" default="true"/>

  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find robmovil_ekf)/launch/rosconsole.conf" if="$(arg debug)"/>

  <!-- Set use_sim_time true for datasets-->
  <param name="use_sim_time" value="true"/>

  <!-- Run clock publisher-->
  <node pkg="vrep_util" type="clock_publisher" name="clock_publisher" output="screen" />

  <!-- Run keyboard input interface-->
  <include file="$(find universal_teleop)/launch/vrep_pioneer_irm.launch" />

  <group if="$(arg detector)">
      <node pkg="imu_laser" type="landmark_detector" name="landmark_detector" output="screen">
        <param name="publish_robot_frame" value="base_link_ekf"/>
        <param name="laser_frame" value="front_laser"/>
      </node>

      <node pkg="imu_laser" type="landmark_detector" name="landmark_detector_gt" output="screen">
        <param name="publish_robot_frame" value="base_link_gt"/>
        <remap from="/landmarks_pointcloud" to="/landmarks_pointcloud/groundtruth"/>
        <remap from="/landmarks" to="/landmarks/groundtruth"/>
      </node>
  </group>

  <node pkg="modelo_diferencial" type="pioneer_odometry" name="pioneer_odometry" output="screen"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_laser" args="0 0 0 0 0 0 base_link laser" />

  <node pkg="robmovil_ekf" type="localizer" name="localizer" output="screen" >
    <param name="only_prediction" type="bool" value="false"/>
  </node>


</launch>
